src/inmanta/protocol/rest/__init__.py

```diff
@@ -220,6 +221,9 @@ async def process(self) -> None:
             arg_type: Optional[type[object]] = self._argspec.annotations.get(arg)
             if arg in self._message:
                 # Argument is parameter in body of path of HTTP request
+                if typing_inspect.is_optional_type(arg_type) and self._properties.operation == "GET":
+                    # Convert Optional[SomeType] to SomeType for get requests
+                    arg_type = more_itertools.one(arg for arg in typing_inspect.get_args(arg_type) if arg is not type(None))
```

Comment:
@arnaudsjs this fails for the `get_agents` call for those argument:

```
    start: Optional[Union[datetime.datetime, bool, str]] = None,
    end: Optional[Union[datetime.datetime, bool, str]] = None,
```
it results in more than one item

Reply:
I reworked that part, I think it is better now.